common/djangoapps/student/roles.py

```diff
@@ -100,17 +145,29 @@ def get_roles(role):
         """
         return ACCESS_ROLES_INHERITANCE.get(role, set()) | {role}
 
+    # TODO: test that this works as expected
+    def get_all_roles_set(self):
+        return self._roles
+
+    def get_roles_by_course_id(self):
+        return self._roles_by_course_id
```

Comment:
The Pythonic way to write a getter is to use a readonly property:

```suggestion
    @property
    def roles_by_course_id(self):
        return self._roles_by_course_id
```

From the perspective of the caller, the property will look like any other field, eg `obj.roles_by_course_id` rather than `obj.get_roles_by_course_id()`

Reply:
I'll do it as per our convention then, but I think generally it's a pretty bad thing to do. There's no benefit other than that it looks slightly nicer, but the drawback is big: when accessing this property, you hide the fact that not just a value is recalled but a function is run, which can lead to bugs.

Reply:
done